src/main/java/dino/task/TaskList.java

```diff
@@ -0,0 +1,117 @@
+package dino.task;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import dino.exception.*;
+
+public class TaskList {
+
+    private final List<Task> taskList;
+
+    public TaskList(List<Task> taskList) {
+        this.taskList = taskList;
+    }
+
+    public List<Task> getTaskList() {
+        return taskList;
+    }
+
+    /**
+     * Adds a new task to the task list
+     *
+     * @param task the task to be added
+     * @return the output message after execution
+     */
+    public String addTask(Task task) {
+        taskList.add(task);
+        int size = taskList.size();
+        String message = "Got it. I've added this task: \n"
+                + "  " + taskList.get(size - 1) + "\n" +
+                "Now you have " + size +
+                (size > 1 ? " tasks" : " task") + " in the list.";
+        return message;
+    }
+
+    /**
+     * Prints out the task list in console, prefixed with index
+     *
+     * @return the output message after execution
+     * @throws EmptyListException if the current task list is empty
+     */
+    public String printTaskList() throws EmptyListException {
+        if (taskList.isEmpty()) throw new EmptyListException();
+        String list = "";
+        for (int i = 0; i < taskList.size(); i++) {
+            list += (i + 1) + ". " + taskList.get(i) + "\n";
+        }
```

Comment:
May I suggest using a StringBuilder instead, since String is immutable in Java, and concatenation creates a new String each time, resulting in O(n^2) time complexity. I think for simple string concatenation, using the '+' operator is fine, but because you are looping through all the tasks in your taskList, I think it can get quite costly.